src/mame/brother/lw350.cpp

```diff
@@ -0,0 +1,1749 @@
+
+#12
+Mitsubishi
+M65020FP
+UA7777-A
+
+#15
+Hitachi
+HD74LS368AP
+
+#10, #11
+Hitachi
+HD74LS157P
+
+#7, #8
+NEC
+D41464C-10
+65,536 x 4-Bit
+Dynamic NMOS RAM
+100 ns
+18-pin plastic DIP
+
+#1
+NEC
+D23C4001EC-172
+UA2849-A
+4MBit Mask ROM
+
+#2
+AMD
+AM27C020
+2BC04
+2MBit (256K x 8-Bit) CMOS EPROM
+
+#3
+Hitachi
+HM65256BSP-10
+32,768 word x 8-bit High Speed Pseudo Static RAM
+100 ns
+DP-28N
+
+#4
+Hitachi HD63266F
+CMOS Floppy Disk Controller
+QFP-64
+Murata Ceralock CST-MXW 16.00 MHz Ceramic Resonator
+
+#5
+Hitachi
+HD6445P4
+CRTC-II (CRT Controller)
+DP-40
+80 system Bus Interface
+4.0 MHz Bus Timing
+
+#9
+Mitsubishi
+M65133FP
+UA7550-A
+
+#14
+Hitachi
+HM62256AF-8
+32,768 word x 8-bit High Speed CMOS Static RAM
+85 ns
+FP-28
+
+720kb Floppy Drive
+
+D-SUB9 CRT Connector
+HSYNC 18.5 kHz
+VSYNC 50 Hz
+PC MDA standard
+  DB9 connector 1+2 GND, 3+4+5 nc, 6 Intensity, 7 Brightness, 8 HSync, 9 VSync
+  HSync positive, VSync negative active
+
+Emulation Status:
+Printer not working
+Floppy Read has some problems (directory working, but LW-450 reports illegal file format when trying to load a .wpt file (content verified with LW-350), but writing seems fine)
+Dictionary ROM probably not correctly mapped
+
+***************************************************************************/
+
+constexpr int MDA_CLOCK = 16'257'000;
+
+class lw450_state : public driver_device
+{
+public:
+	lw450_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		maincpu(*this, "maincpu"),
+		screen(*this, "screen"),
+		palette(*this, "palette"),
+		floppy(*this, "floppy"),
+		fdc(*this, "fdc"),
+		m_crtc(*this, "hd6445"),
+		vram(*this, "vram"),
+		speaker(*this, "beeper"),
+		io_kbrow(*this, "kbrow.%u", 0),
+		rom(*this, "maincpu")
+	{ }
+
+	void lw450(machine_config& config);
+
+private:
+	// devices
+	required_device<z80180_device> maincpu; // use z80180 instead of hd64180rp, because hd64180rf doesn't have hd64180rp's 19-bit address width
+	required_device<screen_device> screen;
+	required_device<palette_device> palette;
+	required_device<lw350_floppy_connector> floppy;
+	required_device<hd63266f_t> fdc;
+	required_device<hd6345_device> m_crtc;
+	required_shared_ptr<uint8_t> vram;
+	required_device<beep_device> speaker;
+	optional_ioport_array<9> io_kbrow;
+	required_region_ptr<uint8_t> rom;
+
+	uint8_t io_72, io_73, io_74, io_75; // gfx
+	uint8_t io_7a, io_b8;
+	uint32_t framecnt;
+
+	uint8_t illegal_r(offs_t offset, uint8_t mem_mask = ~0) {
+		logerror("%s: unmapped memory read from %0*X & %0*X\n", machine().describe_context(), 6, offset, 2, mem_mask);
+		return 0;
+	}
+	void illegal_w(offs_t offset, uint8_t data, uint8_t mem_mask = ~0) {
+		logerror("%s: unmapped memory write to %0*X = %0*X & %0*X\n", machine().describe_context(), 6, offset, 2, data, 2, mem_mask);
+	}
+
+	uint8_t rom72000_r(offs_t offset, uint8_t mem_mask = ~0) {
+		return rom[0x02000 + offset] & mem_mask;
+	}
```

Comment:
Same comments about log spam with debugger windows open and `mem_mask` in 8-bit memory access handlers apply.